src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -334,6 +337,18 @@ RubyBinding binding(Frame callerFrame, Object self, Object[] rubyArgs, RootCallT
                 @Cached(
                         value = "getAdoptedNode(this).getEncapsulatingSourceSection()",
                         allowUncached = true, neverDefault = false) SourceSection sourceSection) {
+
+            final MaterializedFrame materializedCallerFrame = Truffle.getRuntime()
```

Comment:
This is what the materializedCallerFrame looks like in the debugger:
![image](https://github.com/oracle/truffleruby/assets/142719/eb548113-2b8f-4aae-82d9-6e5b2a8b85cb)

Compared to when called from Ruby:
![image](https://github.com/oracle/truffleruby/assets/142719/621e3402-ac48-4d89-8c79-8a29ea6f5668)